src/main/kotlin/com/hibob/academy/types/StoreService.kt

```diff
@@ -0,0 +1,69 @@
+package com.hibob.academy.types
+
+import com.hibob.types.*
+import org.springframework.stereotype.Component
+import java.time.LocalDate
+
+@Component
+class StoreService {
+
+    fun pay(cart: List<Cart>, payment: Payment): Map<String, Check> {
+        return cart.associate {
+            it.clientId to checkout(it, payment)
+        }
+    }
+}
+
+fun fail(message: String): Nothing {
+    throw IllegalStateException(message)
+}
+
+fun customTrueFilter(cart: Cart): List<Product> {
+    return cart.products.filter {product ->
+        product.custom as? Boolean ?: false
+    }
+}
+
+sealed class Statuses {
+    data object SUCCESS : Statuses()
+    data object FAILURE : Statuses()
```

Comment:
anyway if you decide to use object and not enum the style should be Failure (class name)